packages/@aws-cdk/core/lib/cfn-resource.ts

```diff
@@ -310,6 +335,24 @@ export class CfnResource extends CfnRefElement {
     this.dependsOn.add(target);
   }
 
+  /**
+   * Get a shallow copy of dependencies between this resource and other resources
+   * in the same stack.
+   */
+  public obtainResourceDependencies() {
+    return Array.from(this.dependsOn.values());
```

Comment:
why not just return 

`this.dependsOn.values()` here? Is that not a shallow copy?

Reply:
I'm pretty new to TypeScript (and jsii especially) but here's my understanding - please correct me if I'm wrong. Because `this.dependsOn` is a `Set`, we want to convert that into an `Array`, so that jsii can serialize it to other languages. To do that, we grab `.values()` which gets us an iterator, pass that to `Array.from()` which then constructs us a list. This has the happy side-effect of also giving us a 'shallow copy', so we aren't exposing a mutable container directly to a user.